### PR TITLE
examples: add remote example.

### PR DESCRIPTION
**Pull request by pks-t**
This commit implements an example for the remote API. The program mimics behavior of git remote, providing the subcommands add, remove, rename, set-url and show.

### CODE REVIEW

File: examples/remote.c [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+	int retval = 0;
+	struct opts opt = {0};
+	git_repository *repo = NULL;
+
+	parse_subcmd(&opt, argc, argv);
+
+	git_libgit2_init();
+
+	check_lg2(git_repository_open_ext(&repo, ".", 0, NULL),
+		"Could not open repository", NULL);
```
Comment (carlosmn) [inline examples/remote.c]: Why `_open_ext()`? If you want to simulate the git-remote command, it should call `git_repository_discover()` first so it works in subdirectories.

File: examples/remote.c [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+		usage("you need to specify a name and URL", NULL);
+
+	name = o->argv[0];
+	url = o->argv[1];
+
+	check_lg2(git_remote_create(&remote, repo, name, url),
+			"could not create remote", NULL);
+
+	check_lg2(git_remote_save(remote),
+			"could not save remote", NULL);
```
Comment (carlosmn) [inline examples/remote.c]: This call is unnecessary.

File: examples/remote.c [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+	opt->argv = argv + 2;
+}
+
+static void usage(const char *msg, const char *arg)
+{
+	fputs("usage: remote add <name> <url>\n", stderr);
+	fputs("       remote remove <name>\n", stderr);
+	fputs("       remote rename <old> <new>\n", stderr);
+	fputs("       remote set-url [--push] <name> <newurl>\n", stderr);
+	fputs("       remote show [--verbose]\n", stderr);
```
Comment (carlosmn) [inline examples/remote.c]: `[-v | --verbose]`

**Comment by pks-t** [comment]
Applied your comments and rebased onto master.

**Pull request merged**